src/discordtipbot/discordbot.cr

```diff
@@ -122,10 +122,52 @@ class DiscordBot
     end
   end
 
+  private def split(msg : String, spliterator : String, max_length : Int32)
+    msgs = Array(String).new
+
+    # Break up the message into parts which are less than max_length characters each
+    while (msg.size > max_length)
+      # Find the final space before the max_length
+      pos = msg[0, max_length].rindex(spliterator)
+
+      # If there was no space break at the max_length
+      if (pos.nil?)
+        pos = max_length
+      end
+
+      # Record that message!
+      msgs << msg[0, pos]
+
+      # Find the index of the next space after the pos
+      i = msg.index(spliterator, pos)
+
+      if (i.nil?)
+        i = 0
+      else
+        i += 1
+      end
+
+      # If the next space is more than max_length away just record it as the break pos.
+      # FIXME: This will cause problems if a word is longer than max_length! (it is unexpected that
```

Comment:
When making a comment like this, please be more specific as to what "problems" will occur. It takes guesswork and testing for someone later to figure out what you were talking about.